jabgui/src/test/java/org/jabref/gui/copyfiles/CopyMultipleFilesActionTest.java

```diff
@@ -0,0 +1,194 @@
+        databaseContext = mock(BibDatabaseContext.class);
+        filePreferences = mock(FilePreferences.class);
+
+        srcA = Files.createFile(tmpDir.resolve("a.pdf"));
+        srcB = Files.createFile(tmpDir.resolve("b.pdf"));
+        Files.writeString(srcA, "AAA");
+        Files.writeString(srcB, "BBB");
+    }
+
+    private LinkedFileViewModel vmLocal(Path p) {
+        LinkedFileViewModel vm = mock(LinkedFileViewModel.class, RETURNS_DEEP_STUBS);
+        when(vm.getFile().isOnlineLink()).thenReturn(false);
+        when(vm.getFile().findIn(eq(databaseContext), eq(filePreferences))).thenReturn(Optional.of(p));
+        return vm;
+    }
+
+    private LinkedFileViewModel vmOnline() {
+        LinkedFileViewModel vm = mock(LinkedFileViewModel.class, RETURNS_DEEP_STUBS);
+        when(vm.getFile().isOnlineLink()).thenReturn(true);
+        when(vm.getFile().findIn(eq(databaseContext), eq(filePreferences))).thenReturn(Optional.empty());
+        return vm;
+    }
+
+    private LinkedFileViewModel vmMissing() {
+        LinkedFileViewModel vm = mock(LinkedFileViewModel.class, RETURNS_DEEP_STUBS);
+        when(vm.getFile().isOnlineLink()).thenReturn(false);
+        when(vm.getFile().findIn(eq(databaseContext), eq(filePreferences))).thenReturn(Optional.empty());
+        return vm;
+    }
+
+    @Test
+    void executableIsFalseWhenSelectionEmpty() {
+        ObservableList<LinkedFileViewModel> sel = FXCollections.observableArrayList();
+        CopyMultipleFilesAction action = new CopyMultipleFilesAction(sel, dialogService, databaseContext, filePreferences);
+
+        assertFalse(action.executableProperty().get());
+    }
+
+    @Test
+    void executableIsFalseWhenAnyIsOnline() {
+        ObservableList<LinkedFileViewModel> sel = FXCollections.observableArrayList(
+                vmLocal(srcA),
+                vmOnline()
+        );
+        CopyMultipleFilesAction action = new CopyMultipleFilesAction(sel, dialogService, databaseContext, filePreferences);
+
+        assertFalse(action.executableProperty().get());
+    }
+
+    @Test
+    void executableIsFalseWhenAnyLocalButMissing() {
+        ObservableList<LinkedFileViewModel> sel = FXCollections.observableArrayList(
+                vmLocal(srcA),
+                vmMissing()
+        );
+        CopyMultipleFilesAction action = new CopyMultipleFilesAction(sel, dialogService, databaseContext, filePreferences);
+
+        assertFalse(action.executableProperty().get());
+    }
+
+    @Test
+    void executableIsTrueWhenAllLocalAndPresent() {
+        ObservableList<LinkedFileViewModel> sel = FXCollections.observableArrayList(
+                vmLocal(srcA),
+                vmLocal(srcB)
+        );
+        CopyMultipleFilesAction action = new CopyMultipleFilesAction(sel, dialogService, databaseContext, filePreferences);
+
+        assertTrue(action.executableProperty().get());
+    }
+
+    @Test
+    void copiesAllSelectedFilesToChosenDirectory() {
```

Comment:
TempDir can be created for test method only
```suggestion
    void copiesAllSelectedFilesToChosenDirectory(@TempDir Path tmpDir) {
```